app/src/main/java/ch/epfl/sdp/contamination/GridFirestoreInteractor.java

```diff
@@ -37,25 +36,24 @@ String getGridId(Location location) {
 
     public void getTimes(Location location, QueryHandler handler) {
         String path = "LiveGrid/" + getGridId(location) + "/Times";
-        fsi.readCollection(path, handler);
+        super.readCollection(path, handler);
     }
 
     public void readLastLocation(Account account, Callback<Map<String, Object>> callback) {
-        fsi.readDocument("LastPositions", account.getId(), callback);
+        super.readDocument("LastPositions", account.getId(), callback);
     }
 
     public void read(Location location, long time, QueryHandler handler) {
         String path = "LiveGrid/" + getGridId(location) + "/" + time;
-        fsi.readCollection(path, handler);
+        super.readCollection(path, handler);
     }
 
     public void write(Location location, String time, Carrier carrier, OnSuccessListener success, OnFailureListener failure) {
         Map<String, Object> timeMap = new HashMap<>();
         timeMap.put("Time", time);
+        super.writeDocumentWithID("LiveGrid/" + getGridId(location) + "/Times", time, timeMap,
+                success, failure);
```

Comment:
In case of successful operation, the success listener is called 2 times (which may result in unintended behaviors). It could be better to pipeline the two operations and call success at the end

Reply:
Yes I did not see it. I fixed it.